src/Catty/FormulaEditor/ViewController/FormulaEditorViewController.m

```diff
@@ -1085,31 +1135,13 @@ - (IBAction)deleteVariable:(UIButton *)sender {
             }
         }
         if (userVariable) {
-            if(![self isVarOrListBeingUsed:userVariable]) {
+            if(![self isVariableUsed:userVariable] || ![self isListUsed:userVariable]) {
```

Comment:
Please have a closer look at this statement. `isVariableUsed` should only be called if it is a variable, and `isListUsed` only if it is a list.